test/Semantics/resolve88.f90

```diff
@@ -0,0 +1,73 @@
+! RUN: %B/test/Semantics/test_errors.sh %s %flang %t
+! C746, C747, and C748
+module m
+  use ISO_FORTRAN_ENV
+  use ISO_C_BINDING
+
+  ! C746 If a coarray-spec appears, it shall be a deferred-coshape-spec-list and
+  ! the component shall have the ALLOCATABLE attribute.
+
+  type testCoArrayType
+    real, allocatable, codimension[:] :: allocatableField
+    !ERROR: Coarray components must be ALLOCATABLE and have a deferred coshape
+    real, codimension[:] :: deferredField
+    !ERROR: 'pointerfield' may not have the POINTER attribute because it is a coarray
+    !ERROR: Coarray components must be ALLOCATABLE and have a deferred coshape
```

Comment:
It would be nice if the wording of these two messages were similar. For example, the second could be:
`Component 'pointerfield' must have the ALLOCATABLE attribute because it is a coarray`

Reply:
Sounds good.  I'll change it.